### ISSUE TITLE
Fix the way we get the HCI Provider nodes with IBMCloud Baremetal

### ISSUE THREAD

**Issue by yitzhak12** [red-hat-storage/ocs-ci#9374]
Currently, we get the machines with the method `get_machines_by_ips` https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/platform_nodes.py#L3103-L3104:
```
node_ips = [get_node_internal_ip(n) for n in nodes]
return self.ibmcloud_bm.get_machines_by_ips(node_ips)
```
The problem is that we are not using the IPs from the Primary subnet on the nodes (we used them for the first deployment of the first BM cluster, but now we have separated subnet for each cluster, and we are using IPs from this secondary subnet also for the nodes).
In other words, if we run the test on the current clusters, it will not match the IP used by the node (physical server) with the IP assigned to the node from the Primary Subnet (visible in the machine details on IBM Cloud or from the `ibmcloud sl hardware list` command).
So, instead of getting the IBMCloud Baremetal machines using the node IPs, we can get them using the node names.
